core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -54,19 +54,29 @@ def default_address_attributes=(attributes)
       alias_method :ship_address, :default_address
       alias_method :ship_address_attributes=, :default_address_attributes=
 
+      # saves address in address book
+      # sets address to the default if automatic_default_address is set to true
+      # if address is nil, does nothing and returns nil
       def ship_address=(address)
         be_default = Spree::Config.automatic_default_address
-        save_in_address_book(address.attributes, be_default)
+        save_in_address_book(address.attributes, be_default) if address
       end
 
+      # saves order.ship_address and order.bill_address in address book
+      # sets ship_address to the default if automatic_default_address is set to true
+      # sets bill_address to the default if automatic_default_address is set to true and there is no ship_address
+      # if one address is nil, does not save that address
+      # if both do not save, returns nil
```

Comment:
@allisonlarson the return value actually depends completely on the bill address right?  we probably should just not say anything about a return value since it doesn't seem like we return anything super meaningful.